pkg/controller/stack/stack_controller.go

```diff
@@ -207,20 +207,6 @@ func (r *ReconcileStack) Reconcile(request reconcile.Request) (reconcile.Result,
 		}
 	}
 
-	// Run the Pulumi update iff the desired state has not already been
-	// reached, or if it is marked to be deleted.
-	err = sess.getLatestResource(instance, request.NamespacedName)
-	if err != nil {
-		sess.logger.Error(err, "Failed to get latest Stack before updating Stack", "Stack.Name", instance.Spec.Stack)
-		return reconcile.Result{}, err
-	}
-	isStackMarkedToBeDeleted = instance.GetDeletionTimestamp() != nil
-	// Don't run rest of loop if already at desired state, unless marked for deletion.
-	if !isStackMarkedToBeDeleted && (instance.Status.LastUpdate != nil && instance.Status.LastUpdate.State == instance.Spec.Commit) {
-		reqLogger.Info("Stack already at desired state", "Stack.Commit", instance.Spec.Commit)
-		return reconcile.Result{}, nil
-	}
```

Comment:
What was the original argument for short-circuiting here?  I believe it's correct to not do this, but not sure exactly why it was added and thus not sure what potentially undesirable consequence this might have?

Reply:
I am not really sure on this unfortunately. One theory I have is that this was an attempt at avoiding reconciliation loops (modifying the stack CR in the reconciliation loop causes more reconciliations to trigger). #29 introduced this change and was also intended to address the reconciliation loop issue. I am running the integration tests which apparently reproduced the issues so I am hoping we can build confidence with them before pushing this change.